test/libsolidity/util/TestFileParser.cpp

```diff
@@ -169,6 +171,25 @@ vector<solidity::frontend::test::FunctionCall> TestFileParser::parseFunctionCall
 	return calls;
 }
 
+vector<string> TestFileParser::parseFunctionCallEffects()
+{
+	vector<string> result;
+	while (accept(Token::Effect, false))
+	{
+		string effect = m_scanner.currentLiteral();
+		result.emplace_back(effect);
+
+		if (m_scanner.currentToken() == Token::Effect)
+			m_scanner.scanNextToken();
+		if (m_scanner.currentToken() == Token::Newline)
+			m_scanner.scanNextToken();
+		if (m_scanner.currentToken() == Token::EOS)
+			break;
+	}
```

Comment:
Will this correctly parse the following sequence?
```c++
// effectsTest -> 0x2345
// - effectsTest0
// -
// - effectsTest2
// - effectsTest3
// - effectsTest4
```

I think it might stop after the first effect. But I'm not sure which functions here change current token. Does `accept()` scan a new token?